sycl/include/CL/sycl/accessor.hpp

```diff
@@ -426,37 +450,21 @@ class image_accessor
 
   template <int Dims = Dimensions> size_t get_count() const;
 
-  template <> size_t get_count<1>() const {
-    return __invoke_ImageQuerySize<int, OCLImageTy>(MImageObj);
-  }
+  template <> size_t get_count<1>() const { return getCountInternal(); }
   template <> size_t get_count<2>() const {
-    cl_int2 Count = __invoke_ImageQuerySize<cl_int2, OCLImageTy>(MImageObj);
+    cl_int2 Count = getCountInternal();
```

Comment:
What is the value of this indirect call? I think i'm missing something.

Reply:
The getCountInternal function is used in __image_array_slice__, so, I did helper function to avoid code duplication.